### PR TITLE
Tweak iteration logic of HITS

### PR DESCRIPTION
**Pull request by ssmolash**
Refactor HITS convergence iteration logic so that exception message correctly states the number of iterations - iterate while the number of iterations is less than or equal to `max_iter`. If convergence is achieved within `max_iter` iterations, break out of the loop. If convergence is not achieved within `max_iter` iterations, raise a networkx error.

Create test `test_hits_not_convergent` to test for non-convergence of HITS by setting `max_iter` to 0 - test passes when error is raised and test fails when error is not raised.

https://github.com/networkx/networkx/issues/2140

cc: @dschult

### CODE REVIEW

File: networkx/algorithms/link_analysis/hits_alg.py [hunk 1]
```diff
@@ -82,8 +82,8 @@ def hits(G,max_iter=100,tol=1.0e-8,nstart=None,normalized=True):
         s=1.0/sum(h.values())
         for k in h:
             h[k]*=s
-    i=0
-    while True: # power iteration: make up to max_iter iterations
+    i=1
+    while i<=max_iter:  # power iteration: make up to max_iter iterations
```
Comment (jfinkels) [inline networkx/algorithms/link_analysis/hits_alg.py]: This is needlessly complicated; just do `for i in range(max_iter):` then use `max_iter` in the error message below.

File: networkx/algorithms/link_analysis/hits_alg.py [hunk 2]
```diff
@@ -82,34 +82,34 @@ def hits(G,max_iter=100,tol=1.0e-8,nstart=None,normalized=True):
         s=1.0/sum(h.values())
         for k in h:
             h[k]*=s
-    i = 1
-    while i <= max_iter:  # power iteration: make up to max_iter iterations
-        hlast = h
-        h = dict.fromkeys(hlast.keys(), 0)
-        a = dict.fromkeys(hlast.keys(), 0)
```
Comment (chebee7i) [inline networkx/algorithms/link_analysis/hits_alg.py]: These changes are nice to have. We don't go out and change every file, but whenever we do edit a file, PEP8 changes are okay. So if you want to add these back in, please do (separate commit though). Not necessary though.

**Comment by hagberg** [comment]
Needs merge conflict resolution.

**Comment by ssmolash** [comment]
@hagberg: Merge conflicts resolved.

**Pull request merged**